### PR TITLE
Feature/product listing

### PR DESCRIPTION
**Pull request by mintech-dot**

### CODE REVIEW

File: src/components/ui/Button.vue [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+// Computed property for variantClass
+const variantClass = computed(() => {
+  switch (props.variant) {
+    case "primary":
+      return "bg-primary text-white w-full  rounded-[8px] hover:bg-primary/80";
+    case "secondary":
+      return "bg-white text-black rounded-[80px] border-secondary border hover:bg-primary hover:text-white";
+    default:
+      return "";
```
Comment (yazfield) [inline src/components/ui/Button.vue:33]: Default should have proper styling  just with neutral color

File: src/components/ui/Dropdown.vue [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<template>
+    <select
+      id="id"
+      class="bg-white border text-gray-900 text-lg border-secondary/50 w-full sm:w-[250px] rounded-lg p-2.5"
+    >
+      <option v-for="option in options" :key="option.value" :value="option.value">
```
Comment (yazfield) [inline src/components/ui/Dropdown.vue:6]: Should have an extra option for nothingn is selected

File: src/views/ProductsList.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  <div class="px-4 md:px-8 lg:px-24 pt-12 flex justify-between gap-8">
+    <div class="sm:flex md:justify-between gap-8 w-screen">
+      <div class="sm:flex gap-8">
+        <div>
+          <h1 class="text-secondary mt-4">CATEGORIES</h1>
+          <Dropdown :options="productOptions" />
+        </div>
+        <div>
+          <h1 class="text-secondary mt-4">FILTER BY</h1>
+          <Dropdown :options="FilterOptions" />
```
Comment (yazfield) [inline src/views/ProductsList.vue:11]: Filtering is Not part of this ticket but it's okay for now

File: src/App.vue [hunk 4]
```diff
@@ -1,8 +1,7 @@
 <script setup lang="ts">
-import HelloWorld from './components/HelloWorld.vue'
+import products from "./pages/Products.vue";
 </script>
 
 <template>
-  <HelloWorld msg="Welcome to Izar" />
+  <products />
```
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/App.vue:6]: I have another component with the same name. Is that okay?

File: src/api/ApolloClient.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import {
+    ApolloClient,
+    InMemoryCache,
+    createHttpLink,
+  } from "@apollo/client/core";
+  import { DefaultApolloClient } from "@vue/apollo-composable";
+  
+  const httpLink = createHttpLink({
+    uri: "https://readonlydemo.vendure.io/shop-api",
```
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/api/ApolloClient.ts]: Done

File: src/assets/email.svg [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path fill-rule="evenodd" clip-rule="evenodd"
+        d="M4.07675 6.53123C4.23299 6.01907 4.70925 5.64648 5.27254 5.64648H19.2725C19.8358 5.64648 20.3121 6.01907 20.4683 6.53123L12.2725 11.9951L4.07675 6.53123ZM2.52257 6.88275C2.52244 6.89075 2.52242 6.89875 2.52254 6.90674V16.8965C2.52254 18.4153 3.75376 19.6465 5.27254 19.6465H19.2725C20.7913 19.6465 22.0225 18.4153 22.0225 16.8965V6.90671M20.5225 8.29787V16.8965C20.5225 17.5868 19.9629 18.1465 19.2725 18.1465H5.27254C4.58218 18.1465 4.02254 17.5868 4.02254 16.8965V8.29787L11.8565 13.5205C12.1084 13.6885 12.4366 13.6885 12.6886 13.5205L20.5225 8.29787ZM22.0225 6.88278C22.0151 5.3703 20.7868 4.14648 19.2725 4.14648H5.27254C3.75834 4.14648 2.52997 5.37028 2.52257 6.88275"
+        fill="#141718" />
```
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/assets/email.svg:4]: OK

File: src/components/ui/Button.vue (hunk 1 above)
Comment (mintech-dot) [reply src/components/ui/Button.vue:33]: is good like this, or i have to change it ?

File: src/views/ProductsList.vue [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        <input
+          class="border border-secondary/50 rounded-[8px] px-4 py-2 w-full"
+          type="text"
+          placeholder="Search"
+        />
+      </div>
+    </div>
+  </div>
+  <div>
+    <ProductCard :product="products" :loading="loading" :error="error" />
```
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/views/ProductsList.vue:25]: Done

[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/views/ProductsList.vue:25]: Done

File: src/views/ProductsList.vue [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+const productOptions = ref<Option[]>([
+  { value: "laptop", text: "laptop" },
+  { value: "product2", text: "Product 2" },
+  { value: "product3", text: "Product 3" },
+]);
+
+const FilterOptions = ref<Option[]>([
+  { value: "A-z", text: "A-z" },
+  { value: "product2", text: "Product 2" },
```
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/views/ProductsList.vue:50]: I added it only to give a general idea about the user interface of this application

File: src/components/ProductCard.vue [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  <p v-else class="text-center">No products found.</p>
+</template>
+
+<script setup lang="ts">
+import rating from "../assets/rating.svg";
+import Button from "../components/ui/Button.vue";
+import heart from "../assets/heart.svg";
+
+type ProductCardProps = {
+  product: any;
```
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/components/ProductCard.vue]: I have updated the code to use the correct product type from the generated types

File: src/components/ui/Dropdown.vue (hunk 2 above)
Comment (mintech-dot) [reply src/components/ui/Dropdown.vue:6]: Done

File: src/App.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (yazfield) [reply src/App.vue:6]: Yeah they're under different folders, or call it ProductListing if you prefer

File: src/assets/email.svg (hunk 6 above)
[the remark this answers is not in the document]
Comment (yazfield) [reply src/assets/email.svg:4]: Make this a vue component not avg file
You need to have here 'fill="currentColor"' then when you use the component you can change color with text-red-500 for example 

Maybe don't do it to icons that won't have different colors like fb

File: src/assets/vue copy.svg [hunk 10]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="iconify iconify--logos" width="37.07" height="36" preserveAspectRatio="xMidYMid meet" viewBox="0 0 256 198"><path fill="#41B883" d="M204.8 0H256L128 220.8L0 0h97.92L128 51.2L157.44 0h47.36Z"></path><path fill="#41B883" d="m0 0l128 220.8L256 0h-51.2L128 132.48L50.56 0H0Z"></path><path fill="#35495E" d="M50.56 0L128 133.12L204.8 0h-47.36L128 51.2L97.92 0H50.56Z"></path></svg>
```
Comment (yazfield) [inline src/assets/vue copy.svg:1]: Dead file

File: src/views/ProductsList.vue [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        />
+      </div>
+    </div>
+  </div>
+  <div>
+    <div
+    v-if="products && products.items"
+    class="px-4 md:px-8 lg:px-24 pt-12 gap-4 grid xl:grid-cols-3 md:grid-cols-2 sm:grid-cols-1"
+  >
+    <div
```
Comment (yazfield) [inline src/views/ProductsList.vue:29]: This div should be part of the product card, any styling related to the card should be inside product card component, product card should be able to work standalone without the need for product list component.

The v-for then should be over the ProductCard component you don't need another div to do v-for

File: src/views/ProductsList.vue [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  { value: "product3", text: "Product 3" },
+]);
+
+const FilterOptions = ref<Option[]>([
+  { value: "A-z", text: "A-z" },
+  { value: "product2", text: "Product 2" },
+  { value: "product3", text: "Product 3" },
+]);
+
+import { computed } from "vue";
```
Comment (yazfield) [inline src/views/ProductsList.vue:69]: All imports should be in beginning of the file

File: src/views/ProductsList.vue [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+import { useQuery } from "@vue/apollo-composable";
+import { GetProductsDocument, GetProductsQuery } from "../gql/graphql"; 
+
+const { result, loading, error } =
+  useQuery<GetProductsQuery>(GetProductsDocument);
+
+const products = computed(() => result.value?.products) || [];
```
Comment (yazfield) [inline src/views/ProductsList.vue:76]: computed(()=> result.value?.product?.items || [])

This is what I meant earlier

File: src/App.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/App.vue]: Done

File: src/assets/email.svg (hunk 6 above)
[the remark this answers is not in the document]
Comment (mintech-dot) [reply src/assets/email.svg:4]: i made changes in heart.svg

File: src/views/ProductsList.vue (hunk 11 above)
Comment (mintech-dot) [reply src/views/ProductsList.vue]: Done

**Pull request merged**